assets/js/util/date-range/string-to-date.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * String to Date utility.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import { INVALID_DATE_STRING_ERROR } from './constants';
+import { isValidDateString } from './is-valid-date-string';
+
+/**
+ * Converts a valid date string to a Date instance.
+ *
+ * @since n.e.x.t
```

Comment:
In case we include this in the release.
```suggestion
 * @since 1.38.0
```